react/features/video-layout/subscriber.js

```diff
@@ -1,24 +1,103 @@
 // @flow
-
 import {
     VIDEO_QUALITY_LEVELS,
     setMaxReceiverVideoQuality
 } from '../base/conference';
-import { StateListenerRegistry } from '../base/redux';
+import {
+    getPinnedParticipant,
+    pinParticipant
+} from '../base/participants';
+import { StateListenerRegistry, equals } from '../base/redux';
 import { selectParticipant } from '../large-video';
 import { shouldDisplayTileView } from './functions';
+import { setParticipantsWithScreenShare } from './actions';
+
+declare var interfaceConfig: Object;
 
 /**
  * StateListenerRegistry provides a reliable way of detecting changes to
  * preferred layout state and dispatching additional actions.
  */
 StateListenerRegistry.register(
     /* selector */ state => shouldDisplayTileView(state),
-    /* listener */ (displayTileView, { dispatch }) => {
-        dispatch(selectParticipant());
+    /* listener */ (displayTileView, store) => {
+        store.dispatch(selectParticipant());
 
         if (!displayTileView) {
-            dispatch(setMaxReceiverVideoQuality(VIDEO_QUALITY_LEVELS.HIGH));
+            store.dispatch(
+                setMaxReceiverVideoQuality(VIDEO_QUALITY_LEVELS.HIGH));
+
+            _updateAutoPinnedParticipant(store);
         }
     }
 );
+
+/**
+ * For auto-pin mode, listen for changes to the known media tracks and look
+ * for updates to screen shares.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/tracks'],
+    /* listener */ (tracks, store) => {
+        if (typeof interfaceConfig !== 'object'
+            && !interfaceConfig.AUTO_PIN_LATEST_SCREEN_SHARE) {
+            return;
+        }
```

Comment:
Should I being doing this check outside of the function? @paweldomas, @hristoterezov

Reply:
I've heard some rumours a long time ago that the interface config will be included in the state, like the config. If this is our long term goal we will anyway have to do the check inside. But until then I guess it makes sense to be outside. 

But it's not a big deal, so I personally don't have any preferences.